### PR TITLE
asserts: Add assert-string=.

### PR DESCRIPTION
**Pull request by aadcg**

### CODE REVIEW

**Comment by aartaka** [comment]
But then, maybe add more equality operators, to mirror the toolset of CL in general? Not much left: `string-equal`, `char=`, `char-equal`, `=`. And then `/=` and `-not-equal` versions of those...

**Comment by aadcg** [comment]
I don't see why not @aartaka. 

PR revised accordingly.

File: asserts.lisp [hunk 1]
```diff
@@ -50,6 +50,56 @@ vice versa."
...
+  "Assert whether expected and form are CHAR/=."
+  `(expand-assert 'equal-result ,form ,form ,expected ,extras :test #'char/=
+    :full-form ',whole))
+
+(defmacro assert-char-not-equal (&whole whole expected form &rest extras)
+  "Assert whether expected and form are CHAR-NOT-EQUAL."
+  `(expand-assert 'equal-result ,form ,form ,expected ,extras :test #'char-not-equal
+    :full-form ',whole))
+
+(defmacro assert-= (&whole whole expected form &rest extras)
```
Comment (aartaka) [inline asserts.lisp:93]: Note that equal sign and `/=` sign is not preceded by the hyphen in other equality operators/asserts. Maybe remove hyphen?

Comment (aadcg) [reply asserts.lisp:93]: I agree. Perhaps if we'd follow the style we use at Nyxt, we'd do otherwise but...

**Comment by aartaka** [comment]
CC @bobbysmith007, @ryepup, how does that look to you?

**Comment by bobbysmith007** [comment]
Its has been too long to have much insight, but it looks right to me.

**Comment by aartaka** [comment]
Care to merge then? :)

**Comment by aartaka** [comment]
Thanks!

**Comment by Ambrevar** [comment]
There is a mistake in this pull request: none of the new symbols are exported.

@aadcg @bobbysmith007 ?

**Comment by aadcg** [comment]
I will open a new PR. My mistake, sorry.

**Comment by aadcg** [comment]
See #15.

**Pull request merged**